### ISSUE TITLE
Internal error NFUnitCheck.checkUnits: unit check module failed with Block

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#10169]
With this model the unit check module fails with an internal error when pressing "Check model" in OMEdit:
```modelica
model UnitCheckCrash
  Modelica.Blocks.Math.UnitConversions.To_deg to_deg;
equation
  to_deg.u = time;
end UnitCheckCrash;
```

### Steps to Reproduce

Load the model above. Press "Check model". Observe in the log

>[1] 10:27:30 Translation Error
[C:/dev/OM64bit/OMCompiler/Compiler/NFFrontEnd/NFUnitCheck.mo: 83:5-83:91]: Internal error NFUnitCheck.checkUnits: unit check module failed

### Expected Behavior
The unit checking module not crashing, and reporting what the problem is, if any.

### Version and OS
 - OpenModelica Version: v1.21.0-dev-236-gfd1d6d6195
 - OS: Windows 10, 64 bit
 - In Options-Simulation-Additional Translation Settings, I have `-d=nfAPI,nfAPIDynamicSelect`, and `--unitChecking`, if that's relevant.

**Comment by perost** [comment]
The issue is that the unit checker doesn't recognize "deg" as a unit. I added it to the unit table in #10179 (defining it in terms of radians), but it seems like such an obvious thing to do that it makes me think that it was missing for a reason. Maybe @casella knows if there's some reason why it shouldn't be in the unit table?

But that also just fixes the immediate issue with that model, if some other unknown unit were to be used we'd get the same issue again. I'm not sure how it should be handled though, it seems like the unit checker mostly ignores unknown units. But in this case it ends up here when parsing the unit:
https://github.com/OpenModelica/OpenModelica/blob/0954561b67b0246fe7906766ee937ac2da80dbc2/OMCompiler/Compiler/NFFrontEnd/NFUnit.mo#L655-L660

 `unitMul` expects both arguments to be valid units, so `unitMul(inUnit, ut)` fails when `ut` is unknown. I could easily just skip that line in that case, but I'm not sure how it's meant to work.

**Comment by bilderbuchi** [comment]
You'll have the same problem if you use another unit conversion block, e.g. `Modelica.Blocks.Math.UnitConversions.To_degRk`. 
Having non-SI base units as `unit` is contentious I think, but the unit checker should not crash on it at least, maybe ignore it?

**Comment by perost** [comment]
> You'll have the same problem if you use another unit conversion block, e.g. `Modelica.Blocks.Math.UnitConversions.To_degRk`. Having non-SI base units as `unit` is contentious I think, but the unit checker should not crash on it at least, maybe ignore it?

Yes, I think that's the best course of action. I closed my previous PR since it's not clear if we should actually have "deg" in the unit table, and instead just changed the parsing of units in #10180 to not fail in this case and instead return an unknown unit. That seems to work just fine, so I think that's how the unit checker is intended to work.

**Comment by casella** [comment]
> Having non-SI base units as `unit` is contentious I think

Yes. The MLS [Section 19 ](https://specification.modelica.org/master/unit-expressions.html) refers to the SI standard, which allows the use of some non-SI units, such as litre or degree Celsius (or bar, which was however removed from the last edition). I think this should be clarified, in fact there are discussions in the MAP-LANG group about that. The fact that deg is present in the MSL kind of gives some authoritative value to its validity in this context.
> but the unit checker should not crash on it at least,
Absolutely
> maybe ignore it?
In this case it's probably best to consider it, and then issue a warning because you have deg = s in the equation. If the unit is not recognized, a proper message saying so should be issued.

Exactly.

**Comment by perost** [comment]
> > In this case it's probably best to consider it, and then issue a warning because you have deg = s in the equation. If the unit is not recognized, a proper message saying so should be issued.
> 
> I'm not sure if this behaviour was implemented with #10180 , the testcase does not cover the *output+ of `to_deg`, afaict.

Yes, I think the unit checker currently just ignores any unit mismatches involving unknown units. I'm not sure that issuing warnings about them is a good idea currently, since it would probably mean that we'd get a lot of warnings from the MSL.

**Comment by casella** [comment]
> Yes, I think the unit checker currently just ignores any unit mismatches involving unknown units. I'm not sure that issuing warnings about them is a good idea currently, since it would probably mean that we'd get a lot of warnings from the MSL.

I agree we shouldn't get lots of warnings in the MSL, which are solely our fault. I would suggest to add a debug flag to get them _optionally_, so we can work on it when we have the resources and try to improve this.
